draft-ietf-spice-sd-cwt.md

```diff
@@ -579,7 +622,11 @@ kbt-payload = {
 }
 ~~~
 
+The SD-KBT payload MUST contain the audience (`aud`) and issued_at (`iat`) claims.
+It MUST NOT include the issuer (`iss`) or subject (`sub`) claims.
+It MAY include a `cnonce` claim.
 The `cnonce` is a `bstr` and MUST be treated as opaque to the Holder.
+All other claims are OPTIONAL in an SD-KBT.
```

Comment:
its weird that all other claims are optional, but iss and sub are MUST NOT.

the important part is the KBT is signed by key identified in cnf... other things the holder wants to sign are probably use case / protocol specific, and we may want to leave room for them.

Reply:
The point is that the issuer and the subject are the party represented by the cnf (and possibly given an identifier in the sub) in the issued SD-CWT. I want to tell implementers that there is no value sending iss or sub in the KBT so they don't need to waste time writing code checking if they exist and then comparing them to whatever their canonical representation is of the the party in the cnf. Since sub is redactable in the CWT, the language to describe matching rules for iss and sub in the KBT would be super complicated and confusing. (All to manage a possibility that adds zero functionality).

I think we owe it to the community to tell people NO in this case.

Reply:
The holder is the issuer of the key binding token.

However, the iss and sub claims are optional in the key binding token.

Reply:
- put this somewhere:
    "The Holder is the both the subject and the issuer of the Key Binding Token."

- remove the MUST NOT

- in verification section:
      ignore iss or sub for validation purposes (if present). A profile may define additional semantics.

- in security considerations
     warn about attack of issuer or subject in KBT not matching each other or the subject in the SD-CWT (any self-assigned identifier)

Reply:
The Holder is conceptually ... the issuer and subject of the kbt

Reply:
The iss is issuer signed CWT is self asserted by the issuer. the iss and sub in KBT are self asserted by the holder.

Reply:
I just pushed https://github.com/ietf-wg-spice/draft-ietf-spice-sd-cwt/pull/36/commits/233a4fb04eac5352eb2b9cf59f9f8ff6e224e860 which I addresses all these concerns. I would appreciate a look @OR13 at the language in the security considerations section.